packages/nimble-components/src/utilities/type-extensions.d.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * This is a workaround for an incomplete definition of the native dialog element. Remove when using Typescript >=4.8.3.
+ * https://github.com/microsoft/TypeScript/issues/48267
+ * @internal
+ */
+interface HTMLDialogElement {
+    showModal(): void;
+    close(): void;
+}
+
+declare namespace Intl {
+    // roundingPriority has been supported by browsers since 8/23, but TypeScript still hasn't
+    // added it to the type definition. See https://github.com/microsoft/TypeScript/issues/56269
+    interface NumberFormatOptions {
+        roundingPriority?: 'auto' | 'morePrecision' | 'lessPrecision';
```

Comment:
Can we align this to the PR that looks like will add these types? https://github.com/microsoft/TypeScript/pull/56902/files#diff-8cd7de35e3e99beffd3c7f2e8e537e6073b41bc1c10caff2e41be7c97a3c3595R13

```
roundingPriority?: "auto" | "morePrecision" | "lessPrecision" | undefined;
```

Reply:
Sure, but I'm curious: why bother with the redundant `undefined`?

Reply:
You can configure TS to distinguish between optional and assigning undefined: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#exact-optional-property-types---exactoptionalpropertytypes

And based on the other comment doesn't seem worth making an arbitrary difference to avoid conflicts with that type we don't own.